lib/reply.js

```diff
@@ -156,16 +156,17 @@ function pumpCallback (reply) {
 
 function handleError (reply, error, cb) {
   var statusCode = reply.res.statusCode
-  if (error == null) {
-    statusCode = statusCode || 500
-  } else if (error.status >= 400) {
-    statusCode = error.status
-  } else if (error.statusCode >= 400) {
-    statusCode = error.statusCode
-  } else {
-    statusCode = statusCode || 500
+
+  if (error) {
+    if (error.status >= 400) {
+      statusCode = error.status
+    } else if (error.statusCode >= 400) {
+      statusCode = error.statusCode
+    } else if (statusCode === 200) { // default status code
```

Comment:
What if the user sets a `statusCode` and then an error that they did not expect occurs later on? I think it's best to keep the current functionality that sets the status code to  `500` if it is less than `400`.

Reply:
When a user set to 202 and pass an error it's just a programmer error. The current implementation guarantee that an error is treated with 500 as long as you don't overwrite it.

Reply:
StatusCode should be set when the operation can be marked as `2XX` not before when we support something else we do more than we should from my point of view. But I agree that when an error is passed it should has priority.